src/test/java/de/zalando/aruha/nakadi/controller/SubscriptionControllerTest.java

```diff
@@ -0,0 +1,164 @@
+        final Problem expectedProblem = invalidProblem("owning_application", "may not be null");
+        checkForProblem(postSubscription(subscription), expectedProblem);
+    }
+
+    @Test
+    public void whenEventTypesIsEmptyThenUnprocessableEntity() throws Exception {
+        final Subscription subscription = createSubscription("app", ImmutableSet.of());
+        final Problem expectedProblem = invalidProblem("event_types", "size must be between 1 and 1");
+        checkForProblem(postSubscription(subscription), expectedProblem);
+    }
+
+    @Test
+    // this test method will fail when we implement consuming from multiple event types
+    public void whenMoreThanOneEventTypeThenUnprocessableEntity() throws Exception {
+        final Subscription subscription = createSubscription("app", ImmutableSet.of("myET", "secondET"));
+        final Problem expectedProblem = invalidProblem("event_types", "size must be between 1 and 1");
+        checkForProblem(postSubscription(subscription), expectedProblem);
+    }
+
+    @Test
+    public void whenEventTypesIsNullThenUnprocessableEntity() throws Exception {
+        final String subscription = "{\"owning_application\":\"app\",\"consumer_group\":\"myGroup\"}";
+        final Problem expectedProblem = invalidProblem("event_types", "may not be null");
+        checkForProblem(postSubscriptionAsJson(subscription), expectedProblem);
+    }
+
+    @Test
+    public void whenEventTypeDoesNotExistThenNotFound() throws Exception {
+        final Subscription subscription = createSubscription("app", ImmutableSet.of("myET"));
+        when(eventTypeRepository.findByName("myET")).thenThrow(new NoSuchEventTypeException(""));
+
+        final Problem expectedProblem = Problem.valueOf(Response.Status.NOT_FOUND,
+                "Failed to create subscription, event type(s) not found: 'myET'");
+
+        checkForProblem(postSubscription(subscription), expectedProblem);
+    }
+
+    @Test
+    public void whenSubscriptionExistsThenReturnIt() throws Exception {
+        final Subscription subscription = createSubscription("app", ImmutableSet.of("myET"));
+        doThrow(new DuplicatedSubscriptionException("", null)).when(subscriptionRepository).saveSubscription(any());
+
+        final Subscription existingSubscription = createSubscription("app", ImmutableSet.of("myET"));
+        existingSubscription.setId("123");
+        existingSubscription.setStartFrom(Subscription.InitialPosition.BEGIN);
+        existingSubscription.setCreatedAt(new DateTime(DateTimeZone.UTC));
+        when(subscriptionRepository.getSubscription(eq("app"), eq(ImmutableSet.of("myET")), any()))
+                .thenReturn(existingSubscription);
+
+        postSubscription(subscription)
+                .andExpect(status().isOk())
+                .andExpect(content().contentTypeCompatibleWith(APPLICATION_JSON))
+                .andExpect(content().string(sameJSONAs(jsonHelper.asJsonString(existingSubscription))));
```

Comment:
I see that you tested it here. So I think there's no need to test it again in the AT as I commented.